source/common/http/http2/metadata_decoder.cc

```diff
@@ -16,8 +50,12 @@ MetadataDecoder::MetadataDecoder(MetadataCallback cb) : metadata_map_(new Metada
 
   ASSERT(cb != nullptr);
   callback_ = std::move(cb);
+
+  decoder_context_ = std::make_unique<HpackDecoderContext>(*metadata_map_, max_payload_size_bound_);
 }
 
+MetadataDecoder::~MetadataDecoder() = default;
```

Comment:
Optional: move definition to header file.

Reply:
Oh I see.  Thanks for the explanation.

Reply:
Wow, that's amazing. I never would have thought that was "a thing". thanks for the explanation!